Tests/AppTests/ReferenceTests.swift

```diff
@@ -34,6 +34,30 @@ class ReferenceTests: XCTestCase {
         }
     }
 
+    func test_Equatable() throws {
```

Comment:
I'm having second thoughts whether pushing `pathEncoded` all the way into `==` is a good idea. It may be too broad and lead to unrelated issues if users were to have distinct `feature/a` and `feature-a` branches.

It's probably better to restrict this widened equality only to the area of documentation url/target matching.

Will revisit.